nupic/embodied/multitask/algos/custom_mtsac.py

```diff
@@ -196,6 +206,226 @@ def train(self, trainer):
             logging.warn(f"Time to log:             {t4-t3:.2f}")
 
         return np.mean(last_return)
+    
+    def _get_log_alpha(self, samples_data):
+        """Return the value of log_alpha.
+        Args:
+            samples_data (dict): Transitions(S,A,R,S') that are sampled from
+                the replay buffer. It should have the keys 'observation',
+                'action', 'reward', 'terminal', and 'next_observations'.
+        Note:
+            samples_data's entries should be torch.Tensor's with the following
+            shapes:
+                observation: :math:`(N, O^*)`
+                action: :math:`(N, A^*)`
+                reward: :math:`(N, 1)`
+                terminal: :math:`(N, 1)`
+                next_observation: :math:`(N, O^*)`
+        Raises:
+            ValueError: If the number of tasks, num_tasks passed to
+                this algorithm doesn't match the length of the task
+                one-hot id in the observation vector.
+        Returns:
+            torch.Tensor: log_alpha. shape is (1, self.buffer_batch_size)
+        """
+        obs = samples_data["observation"]
+        log_alpha = self._log_alpha
+        one_hots = obs[:, -self._num_tasks:]
+
+        if (log_alpha.shape[0] != one_hots.shape[1]
+                or one_hots.shape[1] != self._num_tasks
+                or log_alpha.shape[0] != self._num_tasks):
+            raise ValueError(
+                "The number of tasks in the environment does "
+                "not match self._num_tasks. Are you sure that you passed "
+                "The correct number of tasks?")
+        
+        with autocast(enabled=self._fp16):
+            return torch.mm(one_hots, log_alpha.unsqueeze(0).t()).squeeze()
```

Comment:
Do we know for sure we need to autocast this operation as well?

Reply:
Yeah we need it because log alpha is computed _with_ gradients tracked when computing the alpha loss